### ISSUE TITLE
cmake can't find /bin/sh during %post container build (rockylinux docker base image)

### ISSUE THREAD

**Issue by ct-clmsn** [apptainer/singularity#6106]
### Version of Singularity:

singularity version 3.8.0+13443-g28df6b6b4

### Expected behavior

I'm attempting to build an application as a %post action. The singularity script installs epel-release, gcc-toolset-10-gcc, gcc-toolset-10-gcc-c++. The build requires downloading and installing cmake-3.21 from the kitware site (the script uses 'wget'). I was expecting cmake to be able to identify and verify '/opt/rh/gcc-toolset-10/root/usr/bin/c++' as a valid compiler. cmake throws an error after detecting the compiler but failing to verify the compiler's validity. It looks like the issue involves '/bin/sh' not being detected properly by cmake. Based on previous singularity issues, this might be related to the base image being a docker container. For clarity, the singularity script starts off like this:

=============================

Bootstrap: docker
From: rockylinux/rockylinux

=============================

### Actual behavior

What actually happend? Why was it incorrect?

The singularity script run perfectly...until cmake executes. I'm able to shell into the container and verify everything seems to be in place. Automating the container and software build process in the container yields the following cmake error:

    -- Check for working CXX compiler: /opt/rh/gcc-toolset-10/root/usr/bin/c++ - broken
    CMake Error at /share/cmake-3.21/Modules/CMakeTestCXXCompiler.cmake:59 (message):
    The C++ compiler
    "/opt/rh/gcc-toolset-10/root/usr/bin/c++"
    is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /tmp/blaze/build/CMakeFiles/CMakeTmp

    Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_5d2d4/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_5d2d4.dir/build.make CMakeFiles/cmTC_5d2d4.dir/build
    gmake[1]: Entering directory '/tmp/blaze/build/CMakeFiles/CMakeTmp'
    gmake[1]: /bin/sh: Command not found
    gmake[1]: *** [CMakeFiles/cmTC_5d2d4.dir/build.make:77: CMakeFiles/cmTC_5d2d4.dir/testCXXCompiler.cxx.o] Error 127
    gmake[1]: Leaving directory '/tmp/blaze/build/CMakeFiles/CMakeTmp'
    gmake: *** [Makefile:127: cmTC_5d2d4/fast] Error 2
    CMake will not be able to correctly generate this project.
    Call Stack (most recent call first):
    CMakeLists.txt:44 (project)
    -- Configuring incomplete, errors occurred!
    See also "/tmp/blaze/build/CMakeFiles/CMakeOutput.log".
    See also "/tmp/blaze/build/CMakeFiles/CMakeError.log".
    make: *** No rule to make target 'install'.  Stop.

### Steps to reproduce this behavior

Write a singularity container script that uses the rockylinux image found on docker's containeryard, install epel-release, gcc-toolset-10-gcc, gcc-toolset-10-gcc-c++, and wget. Write a cmake script for an empty project to check for C++. In %post, have the script wget 'https://github.com/Kitware/CMake/releases/download/v3.21.1/cmake-3.21.1-linux-x86_64.sh' and install it (remember to tell the install script to automatically say 'yes' to the license agreement).  Try to run cmake using the cmake script. The script should fail at the same place.

I can post a link to the singularity script on request. I think this has more to do with pulling a docker container as part of the singularity container build process. There might be an issue with how /bin/sh is symlinked into the container.

### What OS/distro are you running

Ubuntu 20.04.2 LTS Focal Fossa (WSL2)

### How did you install Singularity

From source using the instructions found in INSTALL.md. There was no previous installation of singularity; this rules out previous issues with users installing a new singularity version over an existing singularity version.

**Comment by DrDaveD** [comment]
Can you provide a step-by-step recipe that anyone can use to reproduce this?  The description you gave for reproducing is too vague.  A side benefit is that I often find that when somebody reports a problem they can solve their own problem by going through the effort of finding as simple a reproducer as possible.

**Comment by ct-clmsn** [comment]
Thank you for the response @DrDaveD! please see the singularity build script at this link -> https://github.com/ct-clmsn/miniaturist/blob/main/miniaturist.def

Note, the problem breaks on attempting to run cmake on the git repository cloned for blaze. blaze is a header only library so, it doesn't actually build anything, cmake just seems to be confused as to if /bin/sh exists or not.

**Comment by DrDaveD** [comment]
Well I was able to reproduce the error by downloading the raw miniaturist.def and doing
```
singularity build -f miniaturist.sif miniaturist.def
```
It went ahead and built the .sif and indeed if you run it, there is no /bin/sh.  I don't know why, it would take a lot more debugging.  I suggest that you do some of your own debugging commands to see if /bin/sh is there in the first place and if so what makes it go away.  I added an `ls -l /bin/sh` before the download of cmake and it was there then.  I don't think the cause is something in singularity itself, I think it is something in your .def file.  Yes, after running the cmake installer /bin/sh has disappeared.
